### DISCUSSION TITLE
Running PHPUnit tests in parallel using GitHub actions | Rias.be

### DISCUSSION THREAD

**Discussion by giscus** [riasvdv/rias.be#548] [General]
# Running PHPUnit tests in parallel using GitHub actions | Rias.be

One of our projects has a really large test suite that isn't optimised to run with Laravel's parallel testing. It was starting to take more than 15 minutes to have the whole test suite run inside Github Actions, which is when I searched for a better way to do this.

https://rias.be/blog/running-phpunit-tests-in-parallel-using-github-actions

**Comment by didix16** [comment]
Hi, I've tried your script to run in parallel our tests but it seems they are using the same DDBB and the tests are failing all of them. I get errors of kind 'the table does not exists' or 'the table already exists'. Is there any way that tests don't overlap DDBB each other? Thanks

**Reply by riasvdv** [reply]
You should be setting up the database inside an extra step, for example like we do for the Spatie site: https://github.com/spatie/spatie.be/blob/main/.github/workflows/run-tests.yml#L19-L22

**Reply by didix16** [reply]
Thanks for your answer. Finally we are using the ci_nodes and making database_0. database_1, etc for each job. Thanks @riasvdv!
